File: tech-studio-projects/financial-industry/algo-trading-platform/cpp/orderbook/MatchResult.cpp
#include "MatchResult.h"

// Constructor initializes the result status and details.
MatchResult::MatchResult(bool success, const std::string& details)
    : _success(success), _details(details) {}

// Getter for success status.
bool MatchResult::IsSuccess() const {
    return _success;
}

// Getter for details.
std::string MatchResult::GetDetails() const {
    return _details;
}

File: tech-studio-projects/financial-industry/algo-trading-platform/java/algo-trading-platform/src/com/fintech/algotrading/orderbook/MatchResult.java
package com.fintech.algotrading.orderbook;

public class MatchResult {

    private final boolean success;
    private final String details;

    // Constructor initializes the result status and details.
    public MatchResult(boolean success, String details) {
        this.success = success;
        this.details = details;
    }

    // Getter for success status.
    public boolean isSuccess() {
        return success;
    }

    // Getter for details.
    public String getDetails() {
        return details;
    }
}
